addons/binding/org.openhab.binding.unifi/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,149 @@
+				<label>Password</label>
+				<description>The password to access the UniFi Controller.</description>
+				<context>password</context>
+			</parameter>
+			<parameter name="refresh" type="integer" required="false">
+				<label>Refresh Interval</label>
+				<description>The refresh interval in seconds to poll the UniFi controller</description>
+				<default>10</default>
+			</parameter>
+		</config-description>
+
+	</bridge-type>
+
+	<!-- <thing-type id="wiredClient"> .. coming soon .. </thing-type> -->
+
+	<thing-type id="wirelessClient">
+
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="controller" />
+		</supported-bridge-type-refs>
+		<label>UniFi Wireless Client</label>
+		<description>A wireless client connected to a UniFi wireless network</description>
+
+		<channels>
+			<!-- common wired + wireless client channels -->
+			<channel id="online" typeId="online" />
+			<channel id="site" typeId="site" />
+			<channel id="macAddress" typeId="macAddress" />
+			<channel id="ipAddress" typeId="ipAddress" />
+			<channel id="uptime" typeId="uptime" />
+			<channel id="lastSeen" typeId="lastSeen" />
+			<!-- additional wireless client channels -->
+			<channel id="ap" typeId="ap" />
+			<channel id="essid" typeId="essid" />
+			<channel id="rssi" typeId="rssi" />
+		</channels>
+
+		<representation-property>cid</representation-property>
+
+		<config-description>
+			<parameter name="cid" type="text" required="true">
+				<label>Client ID</label>
+				<description>The MAC address, IP address, hostname or alias of the client</description>
+			</parameter>
+			<parameter name="site" type="text" required="false">
+				<label>Site</label>
+				<description>The site where the client should be found (optional)</description>
+			</parameter>
+			<parameter name="considerHome" type="integer" required="false">
+				<label>Consider Home Interval</label>
+				<description>The interval in seconds to consider the client as home</description>
+				<default>180</default>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="online">
+		<item-type>Switch</item-type>
+		<label>Online</label>
+		<description>Online status of the wireless client</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="site">
+		<item-type>String</item-type>
+		<label>Site Name</label>
+		<description>UniFi Site the client is associated with</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="macAddress">
+		<item-type>String</item-type>
+		<label>MAC Address</label>
+		<description>MAC address of the client</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="ipAddress">
+		<item-type>String</item-type>
+		<label>IP Address</label>
+		<description>IP address of the client</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="uptime">
+		<item-type>Number</item-type>
```

Comment:
You can make this a QuantityType: `Number:Time`.
State would be set `state = new QuantityType<Time>(client.getUptime(), SmartHomeUnits.SECOND);`
(I believe you had some discussion about this on the forum, can't remember if you decided not to use it, so just mention it here)

Reply:
Wanted to focus on getting the binding merged before 2.4 so decided to push the UoM support to the next version.

It's on my list though ;-)